### DISCUSSION TITLE
MindsDB Release notes - August 2021

### DISCUSSION THREAD

**Discussion by C0staTin** [mindsdb/mindsdb#1407] [General]
### Improvements and new features

1. Time Series refactor:
- Reintroduces ARNet
- Adds wrappers for LightGBM and Neural models to support arrays
- Implemented Inferring mode
- Timestamp inferring at group and order by level granularity
2. Added dynamic learning rate to NN.
3. Add model-independent ICP normalizer for dynamic confidence estimation and bounds in numerical, categorical and forecasting tasks
4. Implemented secure connect bundle and protocol version parameters for Cassandra integration.
5. Add option for using custom certificates with MySQL API.
6. New SQL commands in MindsDB MySQL API for SELECT current user and SELECT version and SHOW session status.
7. Added validation to delete datasource action in case it is connected to a model.
8. Added support for joining data tables and models tables using SQL.
9. Added SingleStore integration to the MindsDB Studio.
10. New functionality to allow inline certificates inside config.json.
11. Implemented sample aggregation in MindsDB Mongo API.
12. Increased speed of creation of sql-like and file data sources.
13. Global feature importance was reintroduced in the new version.
14. Add support for uploading files as zip or tar.gz. 
15. New functionality for estimating class probabilities in Neural, LightGBM and Regression models.

### Bug fixes

1. Fix permission error when loading modules.
2. Fix MySQL connection bug when using a custom SSL certificate.
3. Fixed MySQL bug when limit is used in a query.
4. Fix for registering models with the same name when using different integrations.
5. Fixed issues with uploading large datasets.
6. Fix for handling bytes and dicts using Redis Stream.
7. Fix /predict API to allow time series setting.
8. Fix for Apache Superset connection to MySQL API.
9. Fix for `USE GPU` option in MindsDB Studio.
10. Fix for Invalid time value when adding time to a query in MindsDB Studio.
11. Fixed stop training in X seconds functionality.
